compiler/luci/profile/include/luci/Profile/CircleNodeID.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PROFILE_CIRCLE_NODE_ID_H__
+#define __LUCI_PROFILE_CIRCLE_NODE_ID_H__
+
+#include <luci/IR/CircleNode.h>
+
+namespace luci
+{
+
+bool has_node_id(luci::CircleNode *circle_node);
+
+void set_node_id(luci::CircleNode *circle_node, uint32_t id);
```

Comment:
(optional) for `uint32_t id`, can you introduce `CircleNodeID` type? (you can define `CircleNodeID` to any name as you wish) currently we can use as
```cxx
using CircleNodeID = uint32_t;
```

Then IF NEEDED something complicated, we can revise this to
```
struct CircleNodeID 
{
   uint32_t id;
   uint32_t id_2;
   ...
};
```
with this we can reduce code changes.
There are something like this in luci and loco you can reference.
(Although I've not changed existing types in luci as of now...)

If you are sure `uint32_t` is NEVER going to change to any other type, you can go as-is.

Reply:
If then, we may be able to use following implementation :)

```cpp
template <typename T> class CircleNodeID final : public loco::NodeAnnotation
{
public:
  CircleNodeID() = delete;

  CircleNodeID(T node_id) : _node_id{node_id}
  {
    // Do nothing
  }

public:
  T node_id(void) const { return _node_id; }
  // No setter

private:
  T _node_id;
};

} // namespace
```
```cpp
luci::set_node_id<uint32_t>(node, 3);
```

Reply:
Hum... this is not what I thought of.  are you going to use template? if so it would be better to write a full draft code so that we can discuss more.
if not you can use `uint32_t` as ID or use some alias type for ID.

Reply:
Ah, I misunderstood your suggestion.
Then, I will fix like following codes :)

```cpp
using CircleNodeIDType = uint32_t;

class CircleNodeID final : public loco::NodeAnnotation
{
public:
  CircleNodeID() = delete;

  CircleNodeID(CircleNodeIDType node_id) : _node_id{node_id}
  {
    // Do nothing
  }

public:
  CircleNodeIDType node_id(void) const { return _node_id; }
  // No setter

private:
  CircleNodeIDType _node_id;
};
```